generator/src/main/java/com/scottlogic/deg/generator/ValidGenerationEngine.java

```diff
@@ -0,0 +1,35 @@
+package com.scottlogic.deg.generator;
+
+import com.google.inject.Inject;
+import com.scottlogic.deg.generator.decisiontree.DecisionTreeCollection;
+import com.scottlogic.deg.generator.decisiontree.DecisionTreeFactory;
+import com.scottlogic.deg.generator.generation.DataGenerator;
+import com.scottlogic.deg.generator.generation.GenerationConfig;
+import com.scottlogic.deg.generator.outputs.GeneratedObject;
+import com.scottlogic.deg.generator.outputs.targets.FileOutputTarget;
+
+import java.io.IOException;
+import java.util.stream.Stream;
+
+public class ValidGenerationEngine implements GenerationEngine {
+    private final DecisionTreeFactory decisionTreeGenerator;
+    private final DataGenerator dataGenerator;
+
+    @Inject
+    public ValidGenerationEngine(
```

Comment:
This class may be a bit better as ProfileGenerationEngine.
As all of our generation engines should be valid

Reply:
Naming is hard, but...
- `ValidGenerationEngine` implies there is an _invalid_ edition, we've already discussed that name.
- `ProfileGenerationEngine` implies there is a generation engine that doesn't use a profile.

How about `StandardGenerationEngine` or something else that implies data will be generated that abides by the constraints